org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/MavenProjectConfigurator.java

```diff
@@ -37,76 +34,16 @@ public void mavenProjectChanged(MavenProjectChangedEvent[] events, IProgressMoni
   }
 
   private void mavenProjectChanged(MavenProjectChangedEvent event) {
-    IMavenProjectFacade mavenProject = event.getMavenProject();
-    IMavenProjectFacade oldMavenProject = event.getOldMavenProject();
-
-    if (mavenProject == null || oldMavenProject == null) {
-      return;
-    }
-    if (isConfigurationUpdated(mavenProject, oldMavenProject)) {
-      String projectUri = getNormalizedProjectPath(mavenProject);
-      notifyClient(Notifications.UPDATE_PROJECT_CONFIG, projectUri);
-    } else if (!mavenProject
-        .getMavenProjectModules()
-        .equals(oldMavenProject.getMavenProjectModules())) {
-      updateModules(mavenProject, oldMavenProject);
+    if (event.getKind() == MavenProjectChangedEvent.KIND_ADDED) {
+      IProject project = event.getMavenProject().getProject();
+      notifyClient(
+          Notifications.MAVEN_PROJECT_CREATED, ResourceUtils.fixURI(project.getLocationURI()));
```

Comment:
Does it cover that state when pom.xml of existing project was changed? For example when user changes `artifactId` we have to catch this changes and update the name of editor tab and project's description in the project tree.

Reply:
No it does not, however, since the artifact id changes when pom.xml changes, this _should_ be picked up by resource changes. Anyway, we never implemented this feature in the jdt.ls branch anyway, so it's no worse.

Reply:
Refreshing the whole project is overkill for updating the title of the pom file in the editor. We do have code that refreshed the name when a "resource changed" event is sent for a file.